cmd/shortener/server/server.go

```diff
@@ -47,10 +82,17 @@ func NewRouter(options *config.Options, storage *storage.Repository) *chi.Mux {
 		storage: storage,
 	}
 
-	r.Post("/", s.postURLHandle)
-	r.Get("/{id}", s.getShortURLHandle)
-	r.MethodNotAllowed(func(writer http.ResponseWriter, request *http.Request) {
-		writer.WriteHeader(http.StatusBadRequest)
-	})
+	r.Post("/", logger.RequestLogger(GzipMiddleware(s.postURLHandle)))
```

Comment:
CHANGE REQUEST: you should use them as middlewares.

https://github.com/go-chi/chi